lib/bundler/cli/install.rb

```diff
@@ -101,10 +101,27 @@ def run
       # rubygems plugins sometimes hook into the gem install process
       Gem.load_env_plugins if Gem.respond_to?(:load_env_plugins)
 
+      Bundler.plugin("1").manager.call_hook(:before_install)
+      # install plugins before working with others
+      plugin_definition = Bundler.plugin_definition
+
+      if plugin_definition
+        Bundler.ui.info "Setting up plugins..."
+        plugin_definition.validate_ruby!
+
+        Installer.install_plugin(plugin_definition, options)
+        plugin_definition.specs.to_a.each do |spec|
+          next if spec.name == 'bundler'
+          require spec.name
```

Comment:
Only gems with 

`plugin "whatever"`
in the Gemfile

will be required. This is essential for Source plugins to work.